src/main/java/org/opensearch/geospatial/ip2geo/jobscheduler/DatasourceUpdateService.java

```diff
@@ -86,10 +89,36 @@ public void updateOrCreateGeoIpData(final Datasource datasource, final Runnable
             geoIpDataDao.putGeoIpData(indexName, header, reader.iterator(), renewLock);
         }
 
+        waitUntilAllShardsStarted(indexName, MAX_WAIT_TIME_FOR_REPLICATION_TO_COMPLETE_IN_MILLIS);
         Instant endTime = Instant.now();
         updateDatasourceAsSucceeded(indexName, datasource, manifest, fieldsToStore, startTime, endTime);
     }
 
+    /**
+     * We wait until all shards are ready to serve search requests before updating datasource metadata to
+     * point to a new index so that there won't be latency degradation during GeoIP data update
+     *
+     * @param indexName the indexName
+     */
+    @VisibleForTesting
+    protected void waitUntilAllShardsStarted(final String indexName, final int timeout) {
+        Instant start = Instant.now();
+        try {
+            while (Instant.now().toEpochMilli() - start.toEpochMilli() < timeout) {
+                if (clusterService.state().routingTable().allShards(indexName).stream().allMatch(shard -> shard.started())) {
+                    return;
+                }
+                Thread.sleep(SLEEP_TIME_IN_MILLIS);
+            }
+            throw new OpenSearchException(
+                "index[{}] replication did not complete after {} millis",
+                MAX_WAIT_TIME_FOR_REPLICATION_TO_COMPLETE_IN_MILLIS
+            );
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
```

Comment:
please add a test case for scenario when thread execution is interrupted, if that is possible